ui/tests/unit/models/database-connection-test.js

```diff
@@ -0,0 +1,424 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+
+import { AVAILABLE_PLUGIN_TYPES } from 'vault/utils/model-helpers/database-helpers';
+
+// If we refactor the database/connection model we can use this const in the updated form component
+// Ideal scenario be the API provides this schema so we don't have to manually write :)
+const EXPECTED_FIELDS = {
```

Comment:
This could be moved to a test helper file, but decided to hold off unless we plan to use it elsewhere?

I wrote this test coverage before adding the `_filterFields` function to the `database/connection` model